drivers/builtin/src/oid.c

```diff
@@ -9,9 +9,7 @@
 
 #include "common.h"
 
```

Comment:
Now all OID tables are always present in the build. That's a waste of code size. We should guard each function (and the table it consumes) according to where it's used.

For example, `oid_pk_alg` is used for `mbedtls_oid_get_pk_alg()` and `mbedtls_oid_get_oid_by_pk_alg()`. `mbedtls_oid_get_pk_alg()` is only used in `pkparse.c`, guarded by `defined(MBEDTLS_PK_PARSE_C)`, so its definition should also be guarded by `defined(MBEDTLS_PK_PARSE_C)`. Likewise `mbedtls_oid_get_oid_by_pk_alg()` in `pkwrite.c` guarded by `defined(MBEDTLS_PK_WRITE_C)`. And as a consequence the table `oid_pk_alg` should be guarded by `defined(MBEDTLS_PK_PARSE_C) || defined(MBEDTLS_PK_WRITE_C)`.

We can resolve this after 1.0, but I'd prefer to make a reasonable effort now.